### PR TITLE
ludtwig: 0.8.0 -> 0.8.2

### PR DESCRIPTION
**Pull request by MalteJanz**
## Description of changes
https://github.com/MalteJanz/ludtwig/releases/tag/v0.8.1

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by raspher** [comment]
Hi @MalteJanz, it looks like you provided wrong hash for source. You can remove whole hash and run `nix-build -E 'with import <nixpkgs> {}; callPackage ./package.nix {}' -v` to build package locally. The new, correct hash should appear in the logs.

**Review by shyim — APPROVED** [review]

**Comment by raspher** [comment]
Result of `nixpkgs-review pr 285799` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>ludtwig</li>
  </ul>
</details>

**Comment by shyim** [comment]
Good catch, the ofborg failing pipeline is neutral not failing in the checks 🙈

**Comment by MalteJanz** [comment]
Thanks for letting me know. Do you guys have an idea what's wrong here (I'm not very familiar with nix yet 🙈 ).

`Cargo.lock` normally comes from the rust project and there shouldn't be a need to make any changes to build the project as far as I know🤔

**Comment by sikmir** [comment]
Use `cargoPatches`.

**Comment by MalteJanz** [comment]
I just updated ludtwig again (now version 0.8.2) and that should include the latest dependencies and a newly generated `Cargo.lock` (with `cargo generate-lockfile` or `cargo update`).

@sikmir I'm sorry, but what do you mean by `cargoPatches`? As I already said, I'm not very familiar with nix 🙈 .

**Comment by sikmir** [comment]
@ofborg eval

**Comment by sikmir** [comment]
> @sikmir I'm sorry, but what do you mean by `cargoPatches`? As I already said, I'm not very familiar with nix 🙈 .

For more details https://nixos.org/manual/nixpkgs/stable/#building-a-crate-with-an-absent-or-out-of-date-cargo.lock-file

**Comment by shyim** [comment]
I think it's not necessary anymore as it's inside the tag now fixed.

File: pkgs/by-name/lu/ludtwig/package.nix [hunk 1]
```diff
@@ -2,23 +2,23 @@
...
   pname = "ludtwig";
-  version = "0.8.0";
+  version = "0.8.2";
 
   src = fetchFromGitHub {
     owner = "MalteJanz";
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-WF3tEf3SuXiH35Ny4RGLzvEW7yMsFcnVTX52e5qvK5g=";
+    hash = "sha256-iDJvPgBmlca0hJnLrQcGJwlBeX8U1AX4D56gz13ieGg=";
```
Comment (shyim) [inline pkgs/by-name/lu/ludtwig/package.nix:11]:
```suggestion
    hash = "sha256-nNr0iis+wBd+xKJYQL7OWlQnU1DhKztsPHCq3+tX79w=";
```

**Comment by sikmir** [comment]
> I think it's not necessary anymore as it's inside the tag now fixed.

Sure, just for info.

File: pkgs/by-name/lu/ludtwig/package.nix [hunk 2]
```diff
@@ -2,23 +2,23 @@
...
   };
 
   checkType = "debug";
 
-  cargoHash = "sha256-AbT8Jv6v7EVPX5mIplKaBkGrVonA8YWlMvo46coFMzk=";
+  cargoHash = "sha256-f3QNK0YAihOJnCHL34GLpDXl1yt3nCTcyYfM2RocGug=";
```
Comment (shyim) [inline pkgs/by-name/lu/ludtwig/package.nix:16]:
```suggestion
  cargoHash = "sha256-Utho/foZOPz5K3WrOZjAkxvw7+J0RtbW0xvw/Txu/xk=";
```

**Review by shyim — APPROVED** [review]

File: pkgs/by-name/lu/ludtwig/package.nix [hunk 3]
```diff
@@ -2,23 +2,23 @@
...
+    hash = "sha256-nNr0iis+wBd+xKJYQL7OWlQnU1DhKztsPHCq3+tX79w=";
   };
 
   checkType = "debug";
 
-  cargoHash = "sha256-AbT8Jv6v7EVPX5mIplKaBkGrVonA8YWlMvo46coFMzk=";
+  cargoHash = "sha256-Utho/foZOPz5K3WrOZjAkxvw7+J0RtbW0xvw/Txu/xk=";
 
   meta = with lib; {
     description = "Linter / Formatter for Twig template files which respects HTML and your time.";
```
Comment (sikmir) [inline pkgs/by-name/lu/ludtwig/package.nix:19]:
```suggestion
    description = "Linter / Formatter for Twig template files which respects HTML and your time";
```
https://nixos.org/manual/nixpkgs/stable/#var-meta-description

File: pkgs/by-name/lu/ludtwig/package.nix [hunk 4]
```diff
@@ -2,23 +2,23 @@
...
     homepage = "https://github.com/MalteJanz/ludtwig";
     license = licenses.mit;
-    maintainers = with maintainers; [ shyim ];
+    maintainers = with maintainers; [ shyim maltejanz ];
```
Comment (sikmir) [inline pkgs/by-name/lu/ludtwig/package.nix:22]:
```suggestion
    maintainers = with maintainers; [ shyim maltejanz ];
    mainProgram = "ludtwig";
```

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 285799` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ludtwig</li>
  </ul>
</details>

**Comment by MalteJanz** [comment]
@sikmir @shyim can this be merged now or is there anything left to do? 🙈

**Review by shyim — APPROVED** [review]

**Pull request merged**